### PR TITLE
Fix/toggler values

### LINKED ISSUE #2406: Toggler can end up in a state with invalid options
E.g., it can trigger the `Toggler option must exist` error (https://trakt-tv.sentry.io/issues/122932136/?project=4509870926463056)

### PR DESCRIPTION
**Pull request by seferturan**
## 🎶 Notes 🎶

- Fixes #2406
- In some cases the toggler could result in a state where `Toggler option must exist` error was thrown.
- Could not reproduce it locally, but this fix makes it more robust and should counter edge cases.
- Also added tests for it.

### CODE REVIEW

File: projects/client/src/lib/components/toggles/useToggler.ts [hunk 1]
```diff
@@ -17,10 +17,18 @@ export function useToggler<T extends TogglerId, K = TogglerValueMap[T]>(id: T) {
...
+      const stored = safeLocalStorage.getItem(storageKey);
+      if (stored) {
+        const parsed = JSON.parse(stored);
+        if (toggler.options.some((o) => o.value === parsed)) {
+          initialValue = parsed;
+        }
+      }
+    } catch {
+      // ignore parse errors and use default
+    }
```
[the remark this answers is not in the document]
Comment (seferturan) [reply projects/client/src/lib/components/toggles/useToggler.ts]: @gemini-code-assist, good suggestion, done 👌

**Review by vladjerca — APPROVED** [review]
